System/Posix/Env.hsc

```diff
@@ -46,7 +49,7 @@ getEnv ::
 getEnv name = do
   litstring <- withFilePath name c_getenv
   if litstring /= nullPtr
-     then liftM Just $ peekFilePath litstring
+     then Just <$> peekFilePath litstring
```

Comment:
Does this inline the same way as liftM, which has a specialize pragma for IO? https://hackage.haskell.org/package/base-4.14.1.0/docs/src/GHC.Base.html#liftM

Reply:
Well, <$> is infix `fmap`, which is an type-class method, so it should always specialise. Compiling a simple test-case to either Core or STG with -O1 or -O2 yields identical results for both forms.

The 9.0.1 interface file for `Data.Functor` has:
```
476e941a4be6eba1cf16c4abdbf072c4
  (<$>) :: GHC.Base.Functor f => (a -> b) -> f a -> f b
  [HasNoCafRefs, LambdaFormInfo: LFReEntrant 1, Arity: 1,
   Strictness: <S(SL),1*U(1*U,A)>,
   Unfolding: InlineRule (1, True, True)
              (\ @f :: * -> * @a @b ($dFunctor['Many] :: GHC.Base.Functor f) ->
               GHC.Base.fmap @f $dFunctor @a @b)]
```
while in `GHC.Base` we have basically identical definitions for fully saturated `liftM` and `fmap` for IO:
```
af0e0891cc755fd7be45d82375ed39a1
  liftM1 ::
    (a1 -> r)
    -> GHC.Types.IO a1
    -> GHC.Prim.State# GHC.Prim.RealWorld
    -> (# GHC.Prim.State# GHC.Prim.RealWorld, r #)
  [HasNoCafRefs, LambdaFormInfo: LFReEntrant 3, Arity: 3,
   Strictness: <L,1*C1(U)><C(S),1*C1(U(U,U))><L,U>, Inline:,
   Unfolding: InlineRule (3, True, True)
              (\ @a1
                 @r
                 (f['Many] :: a1 -> r)
                 (m1['Many] :: GHC.Types.IO a1)
                 (s['Many] :: GHC.Prim.State# GHC.Prim.RealWorld) ->
               case m1 `cast` (GHC.Types.N:IO[0] <a1>_R)
                      s of ds { (#,#) ipv ipv1 ->
               (# ipv, f ipv1 #) })]

4f16b679e01d41791d2d6fad633b9fcd
  $fFunctorIO2 ::
    (a -> b)
    -> GHC.Types.IO a
    -> GHC.Prim.State# GHC.Prim.RealWorld
    -> (# GHC.Prim.State# GHC.Prim.RealWorld, b #)
  [HasNoCafRefs, LambdaFormInfo: LFReEntrant 3, Arity: 3,
   Strictness: <L,1*C1(U)><C(S),1*C1(U(U,U))><L,U>,
   Unfolding: InlineRule (3, True, True)
              (\ @a
                 @b
                 (f['Many] :: a -> b)
                 (x['Many] :: GHC.Types.IO a)
                 (s['Many] :: GHC.Prim.State# GHC.Prim.RealWorld) ->
               case x `cast` (GHC.Types.N:IO[0] <a>_R) s of ds { (#,#) ipv ipv1 ->
               (# ipv, f ipv1 #) })]
```